keps/sig-windows/4802-windows-node-shutdown/README.md

```diff
@@ -0,0 +1,498 @@
+
+
+#### Alpha Graduation
+
+* Implemented the feature for Windows service only
+
+* Investigate how e2e tests can be implemented (e.g. may need to create fake
+  shutdown event)
+
+#### Alpha -> Beta Graduation
+
+* Addresses feedback from alpha testers
+* Sufficient E2E and unit testing
+
+#### Beta -> GA Graduation
+
+* Addresses feedback from beta
+* Sufficient number of users using the feature
+* Confident that no further API / kubelet config configuration options changes are needed
+* Close on any remaining open issues & bugs
+
+### Upgrade / Downgrade Strategy
+
+Before GA graduation, feature gates will be used to control the enable/disable of this enhacement. During all phase, 
+in order to enable this feature, the kubelet itself will need to run as a windows service.
+
+### Version Skew Strategy
+
+n/a
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `WindowsGracefulNodeShutdown`
+  - Components depending on the feature gate:
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+    No
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+    yes (will require restart of kubelet)
+
+###### Does enabling the feature change any default behavior?
+
+The main behavior change is that during a node shutdown, pods running on the 
+node will be terminated gracefully.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, the feature can be disabled by either disabling the feature gate, or
+      setting `kubeletConfig.ShutdownGracePeriod` to 0 seconds.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Kubelet will attempt to perform graceful termination of pods during a
+        node shutdown.
+
+###### Are there any tests for feature enablement/disablement?
+
+  The e2e framework does not currently support enabling or disabling feature
+  gates. 
+  We have e2e tests to cover the feature when it is enabled and some predefined
+  setting. 
+  Will add node level integration tests when the node level test framework is available for Windows node
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+This fail of rollout will behave like disabling this feature, it wil not impact running workloads during rollout/rollback.
+
+###### What specific metrics should inform a rollback?
+
+n/a
+
+The failure of the roll out will behave like disbling this feature, operators can check the log to get more specific info
```

Comment:
Which logs?  What information will they contain?

Reply:
it should be the default kubelet log, add example for the error information